test/helpers/kubectl.go

```diff
@@ -556,8 +556,14 @@ func (kub *Kubectl) WaitforNPods(namespace string, filter string, minRequired in
 			return false
 		}
 
+		// Determine how many pods we should wait for. We reject pods that are
```

Comment:
This is outside of the scope of this PR, but we should really have a wrapper that says 'wait for deployment to be ready' and introspect the replica count of the deployment. Then we can wait for 2 pods to be ready with the labels from the deployment.

Reply:
Hehe, I considered doing that then I decided that would be too big a change. Still, that is definitely a more reliable approach. In some cases we seem to have an empty filter passed in, and I need to figure out if the intention was "all pods are stable" or "I didn't bother figuring this out and I copied the other test"